compiler/optimizer/src/main/java/edu/snu/nemo/compiler/optimizer/pass/compiletime/reshaping/SailfishRelayReshapingPass.java

```diff
@@ -60,6 +62,21 @@ public SailfishRelayReshapingPass() {
             // before the vertex receiving shuffled data.
             final OperatorVertex iFileMergerVertex = new OperatorVertex(new RelayTransform());
             iFileMergerVertex.getExecutionProperties().put(SkipSerDesProperty.of());
+
+            // Update additional tagged output property of the source vertex if necessary.
```

Comment:
@shpark Can we have an offline meeting to discuss this?
I'd like to know if this something that a certain type of reshaping pass always has to check, and if there's a way to handle this automatically.

Reply:
Due to current implementation that relies on execution property to route additional outputs to different children, we must update the execution property to send data to the inserted vertex, properly. Even if we store information directly to the vertex and edges, without using execution property (similar to `isSideInput()` for runtime edges), we should always update destination information every time we insert a new vertex. In this context, [NEMO-162](https://issues.apache.org/jira/projects/NEMO/issues/NEMO-162) suggest that we provide a high-level api for inserting a new vertex, which can perform these kinds of checks internally